Sources/ProjectDescription/CompatibleXcodeVersions.swift

```diff
@@ -5,55 +5,76 @@ public enum CompatibleXcodeVersions: ExpressibleByArrayLiteral, ExpressibleByStr
     /// The project supports all Xcode versions.
     case all
 
+    case exact(Version)
+
+    case upToNextMajor(Version)
+
+    case upToNextMinor(Version)
+
     /// List of versions that are supported by the project.
-    case list([String])
+    case list([CompatibleXcodeVersions])
 
     // MARK: - ExpressibleByArrayLiteral
 
-    public init(arrayLiteral elements: [String]) {
-        self = .list(elements)
+    public init(arrayLiteral elements: [CompatibleXcodeVersions]) {
+        self = .list(elements.map { $0 })
     }
 
-    public init(arrayLiteral elements: String...) {
-        self = .list(elements)
+    public init(arrayLiteral elements: CompatibleXcodeVersions...) {
+        self = .list(elements.map { $0 })
     }
 
-    enum CodignKeys: String, CodingKey {
+    enum CodingKeys: String, CodingKey {
         case type
         case value
     }
 
     // MARK: - ExpressibleByStringInterpolation
 
     public init(stringLiteral value: String) {
-        self = .list([value])
+        self = .exact("\(value)")
```

Comment:
I find this less surprising
```suggestion
        self = .exact(Version(stringLiteral: value))
```